robot-server/tests/integration/http_api/runs/test_pause_then_cancel.tavern.yaml

```
@@ -0,0 +1,90 @@
+test_name: Test a JSONv6 run can be paused and then cancelled.
+
+marks:
+  - usefixtures:
+      - run_server
+
+stages:
+  - name: Upload a JSONv6 protocol
+    request:
+      url: '{host:s}:{port:d}/protocols'
+      method: POST
+      files:
+        files: 'tests/integration/protocols/wait_then_home_v6.json'
+    response:
+      status_code: 201
+      save:
+        json:
+          protocol_id: data.id
+
+  - name: Create run from protocol
+    request:
+      url: '{host:s}:{port:d}/runs'
+      method: POST
+      json:
+        data:
+          protocolId: '{protocol_id}'
+    response:
+      status_code: 201
+      save:
+        json:
+          run_id: data.id
+
+  - name: Play the run
+    request:
+      url: '{host:s}:{port:d}/runs/{run_id}/actions'
+      method: POST
+      json:
+        data:
+          actionType: play
+    response:
+      status_code: 201
+
+  - name: Pause the run
+    request:
+      url: '{host:s}:{port:d}/runs/{run_id}/actions'
+      method: POST
+      json:
+        data:
+          actionType: pause
+    response:
+      status_code: 201
+
+  - name: Wait for the delay to complete
```

This is referring to the "Pause the run" action above, right?

I might reword this as:

```suggestion
  - name: Wait for the first command - waitForDuration - to complete
```

* `delay` means something very different than `pause` in the Python Protocol API. This isn't the Python Protocol API, but nonetheless, being as tight as possible with terminology might help reduce confusion.
* "Wait for the [delay/pause] to complete" can be misread as "wait until the protocol is no longer [delayed/paused]," but I think what we really mean here is that we want to wait for the pause *action* to be *processed,* so the pause is, in a certain sense, active.

---

@SyntaxColoring it is not, it is waiting for the first command of the protocol (a `waitForDuration`, or a "delay") to complete so that the run ends up paused with its commands in the following state:

1. `waitForDuration` - status: `succeeded`
2. `home` - status: `queued`